src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -2611,6 +2611,103 @@ Task Synchronization API can be used for multiple dynamic synchronization patter
+Signal API is useful to exchange signals between a job and users (via the Workflow Execution portal or the REST API), it cannot be used to send signals from a job to another.
+
+==== Usage
+*Task Signal API* is a *Java API* defined by the following link:../javadoc/org/ow2/proactive/scheduler/signal/SignalApi.html[API documentation].
+
+The Signal API is available inside ProActive Tasks through the *signalapi* <<_glossary_script_bindings,script binding>>.
+
+It is easy to use, and offers a set of basic operations on signals summarized as follows:
+
+  - Ready for signal(s): declares that the current job is ready to receive and handle one or more signals.
+   This operation constructs new signals (by adding the prefix `ready_` to the input signals), then adds them to job signals.
+   For example, if a job declares ready for the signal `soft_stop`, the signal `ready_soft_stop` is added to the set of job signals.
+   Additionally, if the input signal (e.g., `soft_stop`) already exists among the job signals (which means that it has been sent and handled previously),
+    the operation ready for signal automatically removes it, thus the job is ready again to receive and handle the same signal.
+
+  - Send signal(s): adds one or more signals given as input to the current job signals.
+   This operation removes the corresponding ready signals if they already exist. For instance, sending `soft_stop` removes the signal
+   `ready_soft_stop` if it already exists among the job signals.
+
+  - Check for signal(s) reception: verifies whether one or more signals exist among the set of job signals.
+
+  - Wait for signal(s): blocking instruction that waits until one signal (among those given as input) is added to the set of job signals.
+
+  - Remove signal(s): removes one or more signals from the set of job signals.
+
+In addition to the above operations available within ProActive Tasks, users and third-party systems can send signals
+to a specific job via link:/rest/#1218178377[ProActive Signals REST endpoint].
+
+==== Examples
+Two buckets in ProActive *Catalog* contain workflow examples and templates, which illustrate the usage of the Signal API, namely:
+
+* The *Signal* bucket contains many workflow examples that use all signal operations.
+These workflows use a set of predefined groovy scripts stored in the same bucket. Additionally, below we give an example of a groovy code that uses some Signal API operations.
+
+[source, groovy]
+----
+//Declare that the current job is ready to receive and handle the 'soft_stop' signal
+signalapi.readyForSignal("soft_stop")
+
+//Display current job signals
+println("Job signals "+ signalapi.getJobSignals())
+
+//Wait util the 'soft_stop' signal is sent (via the Workflow Execution portal or REST API)
+signalapi.waitFor("soft_stop")
+
+//Remove the 'soft_stop' signal from the set of job signals
+signalapi.removeSignal("soft_stop")
+----
+
+* The *Controls* bucket contains two workflow templates, viz.,:
```

Comment:
I think it's better to talk about the *Controls* template menu in ProActive Studio (same for the Signal menu above) rather than the bucket.

Reply:
Changed as requested